Reorganize Python SDK documentation, add more details

johndmulhausen opened the pull request:
## Summary

This PR reorganizes the Python SDK documentation structure to improve clarity and navigation, added usage details and API positioning/comparison 

### Impact:
- Improved documentation organization with clearer hierarchy
- Better grouping of related SDK documentation
- All existing links preserved through redirects

### Testing:
- All documentation builds successfully
- Internal links have been updated and verified

Fixes documentation structure issues identified in the new reference table of contents project.

<!-- preview-links-comment -->
📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1611#issuecomment-3269042718)**

johndmulhausen committed:
Update references in documentation to correct SDK paths and improve clarity

- Updated links in various guides and reference materials to point to the correct SDK classes and methods.
- Adjusted documentation for artifact and run methods to ensure consistency across the platform.
- Removed outdated automation documentation as part of a cleanup effort.

This commit enhances the accuracy of the documentation and ensures users have the correct references for the W&B SDK.

johndmulhausen committed:
Update system metrics references in documentation to correct SDK paths

- Changed links in various guides and support articles to point to the updated SDK paths for system metrics.
- Removed the outdated system metrics reference page to streamline documentation.

This commit ensures users have accurate and up-to-date references for system metrics in the W&B SDK.

johndmulhausen committed:
Update Python Library title format in documentation

- Changed the title format from "Python Library v(0.21.3)" to "Python Library 0.21.3" for consistency and clarity.

This commit improves the presentation of the Python Library version in the documentation.

ngrayluna committed:
Migrated Ref Quickstert to Dev Quickstart, redirect, removed unused cards in _index.md

johndmulhausen committed:
Update documentation to reflect changes in SDK paths

- Updated references in various guides and support articles to point to the new SDK paths for the W&B API.
- Removed outdated Public API documentation as part of a cleanup effort.

This commit ensures that users have accurate and up-to-date references for the W&B SDK, enhancing the clarity and usability of the documentation.

johndmulhausen committed:
Refactor card formatting in Python documentation

- Updated the card formatting in the Python reference documentation for better consistency and readability.
- Adjusted the weight of the Public API section to improve the order of topics.

This commit enhances the visual structure of the documentation, making it more user-friendly.

johndmulhausen committed:
Add installation and setup instructions to Python SDK documentation

- Introduced a new section detailing the installation process and API key generation for the W&B Python SDK.
- Included code snippets for installing the W&B library and importing the SDK, enhancing user onboarding.

This commit improves the documentation by providing clear steps for new users to get started with the W&B Python SDK.

johndmulhausen committed:
Add installation and setup instructions for W&B Workspaces

- Introduced a new section in the Python SDK documentation covering installation steps and API key generation.
- Included code snippets for installing the W&B library and importing the SDK, along with instructions for specifying the team entity.

This commit enhances the onboarding experience for new users by providing clear and actionable setup guidance.

ngrayluna commented on content/en/ref/python/wandb_workspaces/_index.md:
````diff
@@ -1,17 +1,47 @@
+```python
+TEAM_ENTITY = "<Team_Entity>" # Replace with your team entity
+PROJECT = "my-awesome-project"
+```
````
This is how we currently show users how to import workspace and reports API

```python
import wandb_workspaces.workspaces as ws
import wandb_workspaces.reports.v2 as wr 
```

ngrayluna commented on content/en/ref/python/wandb_workspaces/_index.md:
````diff
@@ -1,17 +1,47 @@
+Install the W&B library.
+
+```
+pip install wandb_workspaces
````
```suggestion
pip install wandb-workspaces
```

ngrayluna commented on content/en/ref/python/wandb_workspaces/_index.md:
```diff
@@ -1,17 +1,47 @@
+
+### Install and import packages
+
+Install the W&B library.
```
```suggestion
Install the W&B Report and Workspaces library.
```

johndmulhausen committed:
Update content/en/ref/python/wandb_workspaces/_index.md

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>

johndmulhausen committed:
Update content/en/ref/python/wandb_workspaces/_index.md

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>

johndmulhausen commented on content/en/ref/python/wandb_workspaces/_index.md:
````diff
@@ -1,17 +1,47 @@
+### Import W&B Python SDK:
+
+```python
+import wandb_workspaces
````
```suggestion
import wandb_workspaces.workspaces as ws
import wandb_workspaces.reports.v2 as wr 
```

johndmulhausen committed:
Update content/en/ref/python/wandb_workspaces/_index.md

ngrayluna committed:
Re added original Quickstart, changed redirect (#1614)

Description
-----------
<!-- Description of the change -->


Related issues
-----------
- Fixes DOCS-12345 <!-- JIRA issue, or remove this line -->
- Fixes #12345 <!-- GitHub issue, or remove this line -->

Co-authored-by: John Mulhausen <john.mulhausen@wandb.com>

johndmulhausen committed:
Update models_quickstart.md to clarify API usage and add dashboard instructions

- Revised API descriptions to specify their usage in the tutorial context.
- Added a new section on viewing training data in the W&B dashboard, enhancing user guidance.
- Corrected links to the Public API for consistency.

This commit improves the clarity and usability of the quickstart guide for new users.

ngrayluna committed:
Reports & Workspaces: Updated _index.md (#1616)

Added bare bones code blocks

Original PR: https://github.com/wandb/docs/pull/1611

To do: Add code examples for creating automations. Use this Colab for
insperation:
https://colab.research.google.com/drive/1spMXk7vjiAp1uFpwDlILihwtipqPzw80?usp=sharing#scrollTo=1ce0401d-bd16-4454-ac9f-a771fc4cff31

<!-- preview-links-comment -->

📄 **[View preview links for changed
content](https://github.com/wandb/docs/pull/1616#issuecomment-3272266951)**

johndmulhausen committed:
Enhance SDK Documentation: Expanded sections on Automations, Custom Charts, Data Types, Functions, and Public API

- Added comprehensive descriptions and examples for the Automations API, detailing triggers, actions, and common use cases.
- Introduced a detailed overview of Custom Charts, including available functions and their applications in model evaluation and data analysis.
- Expanded the Data Types section to cover various media types and their logging capabilities with example usage.
- Provided an overview of global functions in the SDK, highlighting their role in W&B workflows.
- Updated the Public API documentation to clarify its purpose for querying and managing logged data, along with common use cases and example code.

This commit significantly improves the clarity and usability of the SDK documentation, aiding users in effectively utilizing W&B features.

ngrayluna commented on content/en/ref/python/sdk/_index.md:
````diff
@@ -1,7 +1,35 @@
+```python
+TEAM_ENTITY = "<Team_Entity>" # Replace with your team entity
+PROJECT = "my-awesome-project"
+```
````
Specifying an entity or project variable won't do much for the user. 

Let's add how to initialize a run instead? (since it's the atomic unit to W&B?) Maybe either:

```python
import wandb

# Specify your team entity
entity = "<team_entity>"

# Project that the run is recorded to
project = "my-awesome-project"

with wandb.init(entity=entity, project=project) as run:
   run.log({"accuracy": .90, "loss": .10})

```

Or something that actually generates some plots in their dashboard: 

```python
import wandb
import random

wandb.login()

# Specify your team entity
entity = "<team_entity>"

# Project that the run is recorded to
project = "my-awesome-project"

# Dictionary with hyperparameters
config = {
    'epochs' : 10,
    'lr' : 0.01
}

with wandb.init(entity=entity,project=project, config=config) as run:
    # simulating nose for demonstrative purposes
    offset = random.random() / 5

    # Simulate a training run
    for epoch in range(2, config['epochs']):
        acc = 1 - 2**-config['epochs'] - random.random() / config['epochs'] - offset
        loss = 2**-config['epochs'] + random.random() / config['epochs'] + offset
        print(f"epoch={config['epochs']}, accuracy={acc}, loss={loss}")
        run.log({"accuracy": acc, "loss": loss})
```

johndmulhausen replied:
Let's go with option 1 :)

ngrayluna commented on content/en/ref/python/sdk/functions/_index.md:
````diff
@@ -1,6 +1,59 @@
+
+# Finish the run
+wandb.finish()
+```
````
Heads up that we strongly advise not using wandb.finish(). Use context manager instead:

```suggestion
The most common workflow begins with authenticating with W&B, initializing a run, and logging values (such as accuracy and loss) from your training loop:


```python
import wandb

# Authenticate with W&B
wandb.login()

config = {
   "learning_rate": 0.01,
   "epochs": 10,
}

# Project where logs
project = "my-awesome-project"

# Start a new run
with wandb.init(project=project, config=config) as run:
   # Your training code here...
   
   # Log values to W&B
   run.log({"accuracy": acc, "loss": loss})
```

ngrayluna committed:
fixed, added different code examples for data-types/_index.md

johndmulhausen committed:
Update content/en/ref/python/sdk/functions/_index.md

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>

johndmulhausen committed:
Add a paragraph break in Python SDK documentation

johndmulhausen committed:
Merge branch 'main' into new-ref-toc

johndmulhausen committed:
Fix link to Run.define_metric in track log documentation

ngrayluna committed:
Merge branch 'main' of github.com:wandb/docs into new-ref-toc

ngrayluna committed:
fix global functions _index.md

ngrayluna committed:
add working code for custom charts (#1624)

Description
-----------
Removed duplicate info in Automations _index.md, added working code
example for custom-charts/_index.md




<!-- preview-links-comment -->
📄 **[View preview links for changed
pages](https://github.com/wandb/docs/pull/1624#issuecomment-3276434027)**

---------

Co-authored-by: johndmulhausen <john.mulhausen@wandb.com>

ngrayluna committed:
Fix Pydantic formatting: Settings & Automations (#1617)

Fixes formatting for objects defined with Pydantic. 

Note[1]: This was manually copied over from generated docs. 
Note[2]: There are some new args that are being surface, need to check
with Dima and Tony if they should be hidden or exposed (and are just
missing descriptions).

<!-- preview-links-comment -->
📄 **[View preview links for changed
pages](https://github.com/wandb/docs/pull/1617#issuecomment-3273151543)**

---------

Co-authored-by: John Mulhausen <john.mulhausen@wandb.com>

johndmulhausen committed:
Merge branch 'main' into new-ref-toc

johndmulhausen committed:
Update references in documentation to use the new `experiments` namespace for the W&B Python SDK. This includes changes to artifact, run, and logging methods across multiple guides and API references, ensuring consistency and accuracy in the documentation.

johndmulhausen committed:
Update documentation for W&B Public API and Custom Charts

- Revised descriptions for Custom Charts to clarify their programmability and integration with W&B project dashboards.
- Standardized references across multiple Public API documentation files to point to a new include file for usage instructions, enhancing consistency and clarity.

This update improves the overall documentation quality and user understanding of the W&B SDK functionalities.

johndmulhausen committed:
Remove Box3D data type documentation from SDK references

johndmulhausen committed:
Add aliases for system metrics documentation

- Introduced new aliases for the system metrics reference page to improve accessibility and navigation.
- The new aliases include paths for settings and reference guides related to system metrics.

This change enhances the documentation structure and user experience.

johndmulhausen committed:
Update HUGO_VERSION in wrangler.toml to 0.147.8 for compatibility with latest features and improvements.

ngrayluna committed:
see branch name (#1630)

Description
-----------
* Re orders blocks for non Pydantic defined objects
* Adds types for Class properties



<!-- preview-links-comment -->
📄 **[View preview links for changed
pages](https://github.com/wandb/docs/pull/1630#issuecomment-3282378077)**

johndmulhausen committed:
Merge branch 'main' into new-ref-toc

johndmulhausen committed:
Merge branch 'main' into new-ref-toc

johndmulhausen committed:
Update public API documentation to include usage instructions

- Added a reference to a new include file for usage instructions in multiple public API documentation files, enhancing clarity and consistency across the SDK documentation.

johndmulhausen committed:
Update references in documentation to improve clarity and consistency

- Corrected links in multiple files to point to the appropriate sections of the W&B Python SDK, ensuring accurate navigation for users.
- Removed outdated references to the system metrics documentation, streamlining the content and enhancing user experience.

johndmulhausen committed:
Update weights in documentation to improve menu organization

- Adjusted weight values for several sections, including the Python SDK, query panel, release notes, and workspaces API, to enhance the overall navigation structure.
- Removed outdated redirect entries from the _redirects file to streamline the content and improve user experience.

johndmulhausen committed:
Add redirect for Python SDK documentation

- Introduced a new redirect entry in the _redirects file to streamline navigation for users accessing the Python SDK documentation, ensuring a smoother user experience.

johndmulhausen committed:
Merge branch 'main' into new-ref-toc

johndmulhausen commented:
Closing in favor of #1656 which was branched off of this work
